### ISSUE TITLE
Plugin De-Duplicating EC2 Nodes with same name?

### ISSUE THREAD

**Issue by treeves** [rundeck-plugins/rundeck-ec2-nodes-plugin#10]
I have a number of auto-scaling groups that create nodes that, whilst having different EC2 tags, all have the same instance "name".  It seems that the plugin is de-duplicating instances that have the same name and is not displaying them as nodes in Rundeck.   

I.e. when I log in to the EC2 console, I have 330 running instances that come up.  In Rundeck, only 218 appear when I pull with the same key via the EC2 plugin.  

Each ASG that I have for each environment (dev, qa, stage, uat, etc) has multiple instances, all with the same instance name (although different tags, security groups, etc).  Like:
## Name                          Tag

fff-dev1-cq5-stack        AuthorAutoscaleGroup
fff-dev1-cq5-stack        PublishAutoscaleGroup
fff-dev1-cq5-stack        DispatchAutoscaleGroup

But in Rundeck, only one of the above appears. 

Is there something I can do to convince it that different instances are different, even if they have the same name?

**Comment by scalp42** [comment]
Same issue here, with ASG of more than 10 node, only end up seeing 1.

Unfortunately, relying on `instanceID` making it non user friendly :sake:

**Comment by gschueler** [comment]
@scalp42 would allowing something like `tags/Name+instanceId` (combine the two) be sufficient?

**Comment by scalp42** [comment]
yes @gschueler would totally work :+1:

**Comment by scalp42** [comment]
Also, at this point anything would be great but worth mentioning that some folks might have the `+` in the actual name:

```
Allowed characters for all regions are letters, spaces, and numbers representable in UTF-8, plus the following special characters: + - = . _ : / @
```

**Comment by gschueler** [comment]
It definitely makes the mapping syntax more complicated.

`+` allowed in instance names is probably fine, it wouldn't affect the mapping syntax, as the value would be expanded in place.

I was imagining using `+` to append multiple values.  and allowing a quoted value to be a literal string, like:

    nodename.selector=tags/Name+'-'+instanceId

would become "$name-$id" effectively

**Comment by scalp42** [comment]
Oh that makes sense, yeah would totally work!

**Comment by scalp42** [comment]
Also, how can we mix multiple tags in the search ?

```
tags: mesos-slave tags: gz-prod
```

Right now it's acting as an `OR`, is there a way to have it act as a `AND`, meaning only nodes matching **both** tags ?

**Comment by gschueler** [comment]
@scalp42 that was recently filed as an issue as well, see https://github.com/rundeck-plugins/rundeck-ec2-nodes-plugin/issues/41

the workaround is to configure two instances of the ec2 source

**Comment by gschueler** [comment]
@scalp42 sorry i misunderstood.  you mean in the rundeck search? you can search `tags: mesos-slave+gz-prod`

**Comment by scalp42** [comment]
@gschueler got it :)

**Comment by wilddog64** [comment]
does this get resolved in rundeck 2.6.4?
